packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -380,8 +380,6 @@ abstract class ScrollableState<T extends Scrollable> extends State<T> {
   }
 
   void didUpdateScrollBehavior(double newScrollOffset) {
-    if (newScrollOffset == _scrollOffset)
```

Comment:
I audited all the callers and we only call this function when the geometry of the viewport changes.  We'll need to be careful to preserve that invariant going forward to avoid restarting these simulations too often.